internal/server/middleware.go

```diff
@@ -36,6 +36,12 @@ func TimeoutMiddleware(timeout time.Duration) gin.HandlerFunc {
 	}
 }
 
+type RequestContext struct {
```

Comment:
Thoughts on this naming?

I'd like to avoid Context is possible, but I couldn't find another appropriate word for this.  It's easy enough to change later if we find a better name.

Reply:
Context or Ctx is fine. this appears everywhere so shorter is probably better.

Reply:
I don't want to call this `Context` because that is too easily confused with the stdlib context. I was also considering something like `RequestState`, but state is a pretty general term as well.

Reply:
yeah, but you'll see the package name anyway. so something like ctx.Context won't be easily confused.